test/jdk/java/lang/Class/getSimpleName/GetSimpleNameTest.java

```diff
@@ -121,88 +136,91 @@ protected Class<?> findClass(String name) throws ClassNotFoundException {
     }
 
     static class BytecodeGenerator {
-        final String innerName;
-        final String outerName;
+        private static final String CTOR_NAME = "<init>";
+        private static final MethodTypeDesc MD_void = MethodTypeDesc.of(CD_void);
+        final ClassDesc innerName;
+        final ClassDesc outerName;
         final String simpleName;
 
         BytecodeGenerator(String innerName, String outerName, String simpleName) {
-            this.innerName = intl(innerName);
-            this.outerName = intl(outerName);
+            this.innerName = ClassDesc.of(innerName);
+            this.outerName = ClassDesc.of(outerName);
             this.simpleName = simpleName;
         }
 
-        static String intl(String name) { return name.replace('.', '/'); }
-
-        static void makeDefaultCtor(ClassWriter cw) {
-            MethodVisitor mv = cw.visitMethod(ACC_PUBLIC, "<init>", "()V", null, null);
-            mv.visitCode();
-            mv.visitVarInsn(ALOAD, 0);
-            mv.visitMethodInsn(INVOKESPECIAL, "java/lang/Object", "<init>", "()V", false);
-            mv.visitInsn(RETURN);
-            mv.visitMaxs(1, 1);
-            mv.visitEnd();
+        static void makeDefaultCtor(ClassBuilder clb) {
+            clb.withMethodBody(CTOR_NAME, MD_void, ACC_PUBLIC, cb -> {
+                cb.aload(0);
+                cb.invokespecial(CD_Object, CTOR_NAME, MD_void);
+                cb.return_();
+            });
         }
 
-        void makeCtxk(ClassWriter cw, boolean isInner) {
+        void makeCtxk(ClassBuilder clb, boolean isInner) {
             if (isInner) {
-                cw.visitOuterClass(outerName, "f", "()V");
+                clb.accept(EnclosingMethodAttribute.of(outerName,
+                        Optional.of("f"), Optional.of(MD_void)));
             } else {
-                MethodVisitor mv = cw.visitMethod(ACC_PUBLIC | ACC_STATIC, "f", "()V", null, null);
-                mv.visitCode();
-                mv.visitInsn(RETURN);
-                mv.visitMaxs(0, 0);
-                mv.visitEnd();
+                clb.withMethodBody("f", MD_void, ACC_PUBLIC | ACC_STATIC,
+                        CodeBuilder::return_);
             }
         }
 
         byte[] getNestedClasses(boolean isInner) {
-            String name = (isInner ? innerName : outerName);
-            ClassWriter cw = new ClassWriter(0);
-            cw.visit(V1_7, ACC_PUBLIC + ACC_SUPER, name, null, "java/lang/Object", null);
-
-            cw.visitInnerClass(innerName, outerName, simpleName, ACC_PUBLIC | ACC_STATIC);
-
-            makeDefaultCtor(cw);
-            cw.visitEnd();
-            return cw.toByteArray();
+            var name = (isInner ? innerName : outerName);
+            return Classfile.build(name, clb -> {
+                clb.withSuperclass(CD_Object);
+                clb.withFlags(AccessFlag.PUBLIC, AccessFlag.SUPER);
+                clb.accept(InnerClassesAttribute.of(
```

Comment:
During the API discussions there was slightly more recommended to use `ClasfileBuilder::with` over `ClasfileBuilder::accept`, however it is just a cosmetic difference.